test/refactoring/ConvertAsync/convert_pattern.swift

```diff
@@ -5,6 +5,305 @@ enum E : Error { case e }
+  // MIXED-TUPLE-RESULT3-NEXT:   print("boo")
+  // MIXED-TUPLE-RESULT3-NEXT: }
+
+  // RUN: %refactor-check-compiles -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):3 | %FileCheck -check-prefix=MULTIPLE-TUPLE-PARAM %s
+  multipleTupleParam { strs, ints, err in
+    guard let (str1, str2) = strs, let (int1, int2) = ints else {
+      print("ohno")
+      return
+    }
+    print(str1, str2, int1, int2)
+  }
+  // MULTIPLE-TUPLE-PARAM:      do {
+  // MULTIPLE-TUPLE-PARAM-NEXT:   let ((str1, str2), (int1, int2)) = try await multipleTupleParam()
+  // MULTIPLE-TUPLE-PARAM-NEXT:   print(str1, str2, int1, int2)
+  // MULTIPLE-TUPLE-PARAM-NEXT: } catch let err {
+  // MULTIPLE-TUPLE-PARAM-NEXT:   print("ohno")
+  // MULTIPLE-TUPLE-PARAM-NEXT: }
+
+  // RUN: %refactor-check-compiles -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):3 | %FileCheck -check-prefix=MULTIPLE-TUPLE-PARAM2 %s
+  multipleTupleParam { strs, ints, err in
+    guard let (str1, str2) = strs, var (int1, int2) = ints else {
+      print("ohno")
+      return
+    }
+    print(str1, str2, int1, int2)
+  }
+  // MULTIPLE-TUPLE-PARAM2:      do {
+  // MULTIPLE-TUPLE-PARAM2-NEXT:   var ((str1, str2), (int1, int2)) = try await multipleTupleParam()
+  // MULTIPLE-TUPLE-PARAM2-NEXT:   print(str1, str2, int1, int2)
+  // MULTIPLE-TUPLE-PARAM2-NEXT: } catch let err {
+  // MULTIPLE-TUPLE-PARAM2-NEXT:   print("ohno")
+  // MULTIPLE-TUPLE-PARAM2-NEXT: }
+
+  // RUN: %refactor-check-compiles -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):3 | %FileCheck -check-prefix=MULTIPLE-TUPLE-PARAM3 %s
+  multipleTupleParam { strs, ints, err in
+    guard let (str1, str2) = strs, let (int1, int2) = ints else {
+      print("ohno")
+      return
+    }
+    print(strs!)
+    print(str1, str2, int1, int2)
+  }
+  // MULTIPLE-TUPLE-PARAM3:      do {
+  // MULTIPLE-TUPLE-PARAM3-NEXT:   let (strs, (int1, int2)) = try await multipleTupleParam()
+  // MULTIPLE-TUPLE-PARAM3-NEXT:   let (str1, str2) = strs
+  // MULTIPLE-TUPLE-PARAM3-NEXT:   print(strs)
+  // MULTIPLE-TUPLE-PARAM3-NEXT:   print(str1, str2, int1, int2)
+  // MULTIPLE-TUPLE-PARAM3-NEXT: } catch let err {
+  // MULTIPLE-TUPLE-PARAM3-NEXT:   print("ohno")
+  // MULTIPLE-TUPLE-PARAM3-NEXT: }
+}
+
+// RUN: %refactor-check-compiles -convert-to-async -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=NAME-COLLISION %s
+func testNameCollision(_ completion: () -> Void) {
+  let a = ""
+  stringTupleParam { strs, err in
+    guard let (a, b) = strs else { return }
+    print(a, b)
+  }
+  let b = ""
+  stringTupleParam { strs, err in
+    guard let (a, b) = strs else { return }
+    print(a, b, strs!)
+  }
+  print(a, b)
+  completion()
+}
+
+// It's a little unfortunate that the second 'strs' gets renamed to 'strs1'
+// given the first one gets replaced with an inline pattern, but I don't expect
+// this to be a common occurrance.
```

Comment:
I assume this is just because we're currently adding all the success param names no matter what. You could probably pull the `ToPrint` construction out of `addAwaitCall` and then use it in `prepareNames` instead. Or merge the two/something else. Up to you though, probably fine as is anyway 🤷

Reply:
Fixed in latest update